builtInFunctions/interface.go

```diff
@@ -3,5 +3,6 @@ package builtInFunctions
 // CrossChainTokenCheckerHandler should check if token is from another chain/sovereign shard
 type CrossChainTokenCheckerHandler interface {
 	IsCrossChainOperation(tokenID []byte) bool
+	IsAllowedToMint(address []byte, tokenID []byte) bool
```

Comment:
it is better to have IsAllowedToExecute and use it for burn as well.

Reply:
Right, that's a good suggestion. Refactored.